manual/src/tutorials/moduleexamples.etex

```diff
@@ -80,47 +72,39 @@ concerned expression. This can also make the code both easier to read
 (since the code fragment is more self-contained).
 Two constructions are available for this purpose:
 \begin{caml_example}{toplevel}
-let open PrioQueue in
-insert empty 1 "hello";;
+let open Fifo in
+add "hello" empty;;
 \end{caml_example}
 and
 \begin{caml_example}{toplevel}
-PrioQueue.(insert empty 1 "hello");;
+Fifo.(add "hello" empty);;
 \end{caml_example}
 In the second form, when the body of a local open is itself delimited
 by parentheses, braces or bracket, the parentheses of the local open
 can be omitted. For instance,
 \begin{caml_example}{toplevel}
-PrioQueue.[empty] = PrioQueue.([empty]);;
-PrioQueue.[|empty|] = PrioQueue.([|empty|]);;
-PrioQueue.{ contents = empty } = PrioQueue.({ contents = empty });;
-\end{caml_example}
-becomes
-\begin{caml_example}{toplevel}
-PrioQueue.[insert empty 1 "hello"];;
+Fifo.[empty] = Fifo.([empty]);;
+Fifo.[|empty|] = Fifo.([|empty|]);;
+Fifo.{ contents = empty } = Fifo.({ contents = empty });;
 \end{caml_example}
 This second form also works for patterns:
 \begin{caml_example}{toplevel}
 let at_most_one_element x = match x with
-| PrioQueue.( Empty| Node (_,_, Empty,Empty) ) -> true
+| Fifo.{ front = ([] | [_]); rear = [] } -> true
```

Comment:
This can be more simply written using ordinary label qualification `{ Fifo.front = ...; rear = ... }`, but I guess it is OK for the purposes of illustrating the local open in patterns.

Reply:
Yes, the original example (with the prio queue) was also intended to illustrate local open in patterns.  Also, I prefer `Fifo.{ ... }` (a local open for the whole record) to the trick of qualifying one record field and not the others.